File: c言語/Pythonで学ぶアルゴリズムの教科書 一生モノの知識と技術を身につける/chapter2_プログラミングの力を養う/5.nの階乗を求める。/factorial.c
#include <stdio.h>

int factorial_sample()
{
	// 10の階乗を求める。
	int total = 10;
	for( int ii = 9; 0 < ii; ii-- ) {
		total *= ii;
	}
	printf("10!：%d", total );

	return 0;
}

int main(void)
{
	factorial_sample();	// 10!：3628800

	return 0;
}

// vim: set ts=4 sts=4 sw=4 tw=0 ff=unix fenc=utf-8 ft=c noexpandtab:

File: Java言語/Pythonで学ぶアルゴリズムの教科書 一生モノの知識と技術を身につける/chapter2_プログラミングの力を養う/5.nの階乗を求める。/factorial.java
class factorial {
	static void recursionsample(){
		// 10の階乗を求める。
		int total = 10;
		for( int ii = 9; 0 < ii; ii-- ) {
			total *= ii;
		}
		System.out.printf("10!：%d\n", total);
	}

	static int factorialsample(int count){
		if( count > 0 ) {
			System.out.printf("再帰関数%2d回目の実行\n", count);
			factorialsample( count - 1 );
		}

		return count;
	}

	static int factorial1(final int MAX){
		// 階乗を求める関数。
		//	この関数は、サンプル関数に引数を受け取る形にしただけのもの。
		int count = 1;
		for( int ii = MAX; 0 < ii; ii-- ) {
			count *= ii;
		}

		return count;
	}

	public static void main(String[] args){
		int factorial = factorial1(10);
		System.out.printf("10!：%d\n", factorial);	// 10!：3628800

		recursionsample();	// 10!：3628800
		factorialsample(10);	// 再帰
	}
}

// vim: set ts=4 sts=4 sw=4 tw=0 ff=unix fenc=utf-8 ft=java noexpandtab:
